### ISSUE TITLE
Feature request: group.by + split.by functionality to do_ViolinPlot.

### ISSUE THREAD

**Issue by AlicePsyche** [enblacar/SCpubr#5]
Hi, Just found this very useful package. Especially `do_ViolinPlot` !! (it is a disaster to use the seurat's VlnPlot if my obj has more than 100K cells). But it seems that it lacks the parameters that Seurat has, such as `split.by`. Would be wonderful if you may have time to add them in the future. Great thanks!

**Comment by enblacar** [comment]
Hi @AlicePsyche,

Thanks for using the package! I am happy that you find it useful!

`split.by` in `SCpubr::do_ViolinPlot()` is a feature that will be added to `SCpubr` in the next update! 

However, if you **`REALLY`** need it right now, I just recently added it in the last commit, but bear in mind that this is a development version and it is `prone to errors`, so please download again the `official release from CRAN` once you get your plot done!

Install the commit using:
```r
devtools::install_github("enblacar/SCpubr", ref = "59895b45ff3b412d7e467a12146acec73c797c81")
```

Also tell me if `split.by` is not working as intended (boxplots will not be plotted on top of the split violins, I did not manage to make them look good).

Best,
Enrique

**Comment by w2felix** [comment]
Hi, just a short addition here:
I have many cell types and if you have more than one feature of interest added, it would be great to set the **ncol** to a different value.
In addition, the rotation of the **x-axis labels to 90°** would also be great, as with 45° often the labels on the left are cropped.
Thanks!

**Comment by enblacar** [comment]
Hi @w2felix,

Thanks for the feedback!

As of now, `do_ViolinPlot()` works independent of Seurat, in the sense that I am not relying on `Seurat::do_VlnPlot()` to do the plotting. This also means that, when used, I am expecting the user to provide a single feature. Anyway, seeing the interest in plotting multiple features at the same time, I will give it some more work!

Completely agree on the rotation of the labels, I should modify it to allow 0, 45 and 90 degrees. I will add it to the ToDo list!

Best,
Enrique

**Comment by enblacar** [comment]
Hi @w2felix,

I made some changes to `do_ViolinPlot()` to allow for:
- Multiple features.
- `ncol` parameter to control the number of columns of the resulting plot.
- Reproduced the `same.y.lims` behaviour in Seurat. If it is set to `TRUE`, it will force all violin plots to share the same Y axis (the range is the max and min values across all features).

Can you install the latest commit and give it a go and tell me if this is what you wanted?

```r
devtools::install_github("enblacar/SCpubr", ref = "124303d0c15d789099cfd3875f069f59e750750a")
```

Again, remember that this is a development version and should be used with caution. Once you check it out, please download the CRAN version back. All these changes will come in a CRAN update soon.

Best,
Enrique

**Comment by enblacar** [comment]
Hi @AlicePsyche @w2felix,

I am closing this issue as I believe all the necessary changes have been implemented in [SCpubr v1.0.2](https://github.com/enblacar/SCpubr/blob/main/NEWS.md). 

Please update the package from `CRAN` and if some issues persist, feel free to open another Issue! 

Thanks for your feedback!
Enrique
